chainstate/src/detail/mod.rs

```diff
@@ -74,7 +74,8 @@ pub enum BlockSource {
 }
 
 impl Chainstate {
-    pub fn wait_for_all_events(&self) {
+    #[allow(dead_code)]
+    pub(crate) fn wait_for_all_events(&self) {
```

Comment:
There's no need to specify pub(crate) instead of pub because the detail module is not reexported to the outside. Please remove them. The use of pub(crate) should be reserved for very special cases. Usually it's used as the last line of defense when there's a design issue. We don't have that here.